frontend/src/routes/(unauthenticated)/login/+page.svelte

```diff
@@ -76,6 +77,8 @@
             autocomplete="current-password"
           />
 
+          <PasswordStrengthMeter password={$form.password} />
```

Comment:
We'll want to measure and save password strength in other places too right?
- Register page
- Reset password
- Admin's Edit user modal

Reply:
Done in commit 74f8a57b.

Reply:
You added the strength-meter in more places, but not recording the password strength (i.e. calling `UpdatePasswordStrength`). I think doing it when a user registers/gets registered is critical. But, we might as well be thorough right? What if an admin resets the password to something bad and then tries to make the user a manager.

Reply:
I'm not a huge fan of storing password strength in the database based solely on what the frontend tells us is the password strength. For example, the register API currently passes a password hash to the backend, not a plaintext password. We can't measure the password strength of a hash without actually attempting to crack the hash, which is not something we want to do. So there are three options:

1. Change the register API so that the frontend passes plaintext passwords to the backend, which (of course) hashes them before storing them in the database. Then we can run the Zxcvbn.NET library on the plaintext password, and get a strength score that we know is valid.
2. Add a password-strength score to the RegisterAccountInput record, so that the registration page will submit its opinion of the password strength, and the backend will trust that the API call hasn't been tampered with or artificially created to give a user with password "12345678" a strength score of 4 (the max, it's on a 0-4 scale).
3. Do not save a password strength when we register a new account. Then the column is NULL in the database, which means that when that user first logs in with that password, the strength score will be calculated (by the backend) and saved.

I went with option 3 for the register page.

Likewise, when an admin resets a user's password, the frontend only passes a hash to the backend. So I chose to [reset the password-strength column](https://github.com/sillsdev/languageforge-lexbox/commit/5390626ad2d38d89fb3423414d3a1137a1dea24f) when the admin resets a password, so that it will be calculated and set again when the user first logs in. This is also option 3.

Reply:
I've added commit 57b0e0f0 to make it more clear that the registration page does not store password strength in the database.

I'll leave this conversation unresolved so you can more easily find it to add your comments. Please go ahead and resolve it if you end up agreeing with my decision, otherwise we can hash it out in Slack so that Kevin can chime in.

Reply:
@rmunn You're right, it's a tad silly accepting the password strength from the client. On the other hand 🤔 who would tamper with that? Power users?

Ultimately it depends on how reliable we want this strength info to be.
The way admins work today, if we don't get a password strength score on the register page, it's very likely that they'll add users to projects before we have password strengths for them.

@hahn-kev What do you think?

Reply:
So here's the thing, we want to move all users to using JWT for S&R, which means we won't always have the password in the clear, but we still need to record the strength somewhere.

I think we're fine with trusting the strength from the client. There's 2 cases where an attacker might abuse that and neither make any sense. 

1. attacker creates their own account by manually calling our api and lying about the strength. This only matters if the attacker somehow gets to be promoted to admin, if we're doing that we already lost and it doesn't matter if they lied about the password strength.
2. attacker somehow resets an admin password and when they do that lie about the strength. This doesn't make any sense, if they are able to reset an admin password they could just as likely use a strong password, either way password strength checks isn't going to help us here.

The whole point of this is to help guard us from giving users permission when their account is at risk due to a poorly secured password. 

I say the frontend reports the strength to the backend and we just trust it.

Reply:
Yes, I also think the advantages of trusting the frontend outweigh the risks.

Reply:
Done in commit 70ec2b18.